python/lsst/pipe/tasks/makeCoaddTempExp.py

```diff
@@ -501,26 +505,15 @@ def getCalibratedExposure(self, dataRef, bgSubtracted):
             mi += background.getImage()
             del mi
 
-        if self.config.doApplyUberCal:
-            if self.config.useMeasMosaic:
-                from lsst.meas.mosaic import applyMosaicResultsExposure
-                # NOTE: this changes exposure in-place, updating its Calib and Wcs.
-                # Save the calibration error, as it gets overwritten with zero.
-                calibrationErr = exposure.getPhotoCalib().getCalibrationErr()
-                try:
-                    applyMosaicResultsExposure(dataRef, calexp=exposure)
-                except dafPersist.NoResults as e:
-                    raise MissingExposureError('Mosaic calibration not found: %s ' % str(e)) from e
-                photoCalib = afwImage.PhotoCalib(exposure.getPhotoCalib().getCalibrationMean(),
-                                                 calibrationErr,
-                                                 exposure.getBBox())
-            else:
-                photoCalib = dataRef.get("jointcal_photoCalib")
-                skyWcs = dataRef.get("jointcal_wcs")
-                exposure.setWcs(skyWcs)
+        if self.config.doApplyExternalPhotoCalib:
+            photoCalib = dataRef.get(f"{self.config.externalPhotoCalibName}_photoCalib")
         else:
             photoCalib = exposure.getPhotoCalib()
 
+        if self.config.doApplyExternalSkyWcs:
+            skyWcs = dataRef.get(f"{self.config.externalSkyWcsName}_wcs")
+            exposure.setWcs(skyWcs)
```

Comment:
I am belatedly seeing the comment in the **obs_subaru HSC** makeCoaddTempExp.py which states "This was copy-pasted from MakeCoaddTempExpTask, and extra step added to check whether an additional mask is needed." BOOO. The docstring further admits it is a hack.

This explains why I was confusing myself thinking I had already reviewed this code 😠 

Anyway, assuming this code duplication is an independent problem I cannot control, and will be fixed in DM-20074 per the comment in obs_subaru, please see my review comments in obs_subaru, where I suggest changes in literally identical code.